src/api/db/data/20230221000000_fix_migration_version.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class DummySchemaMigration < ApplicationRecord
+  self.table_name = 'schema_migrations'
+end
+
+# This data migration fixes a wrong migration's version number introduced in the `schema_migrations` table.
+# Assuming that `rails db:migrate:with_data` will alternate schema and data migrations following the date order,
+# we name this file with an old date. This way we force this to run before any other schema migration.
+
+class FixMigrationVersion < ActiveRecord::Migration[7.0]
+  def up
+    wrong_version = DummySchemaMigration.find_by(version: '202302224114624')
+    wrong_version&.update(version: '20230224114624')
```

Comment:
In production, the version will be there. In new instances it won't, that's why we do the update only if it's present.